controllers/machinedeployment_controller.go

```diff
@@ -111,7 +111,8 @@ func (r *MachineDeploymentReconciler) reconcile(ctx context.Context, d *clusterv
 		if d.Status.ObservedGeneration < d.Generation {
 			patch := client.MergeFrom(d.DeepCopy())
 			d.Status.ObservedGeneration = d.Generation
-			if err := r.Client.Status().Patch(ctx, d, patch); err != nil {
+			// Patch using a deep copy to avoid overwriting any unexpcted Spec/Metadata changes from the returned result
+			if err := r.Client.Status().Patch(ctx, d.DeepCopy(), patch); err != nil {
```

Comment:
Looking at this in more detail now - is this deep copy necessary, given that we return immediately following this?

Reply:
@ncdc yes, you are correct, and we also shouldn't be requeueing the request if it has an empty label selector, either.